examples/lock-app/lock-common/src/LockEndpoint.cpp

```diff
@@ -475,15 +506,57 @@ bool LockEndpoint::setLockState(const Nullable<chip::FabricIndex> & fabricIdx, c
         "Lock App: specified PIN code was found in the database, setting door lock state to \"%s\" [endpointId=%d,userIndex=%u]",
         lockStateToString(lockState), mEndpointId, userIndex);
 
-    mLockState                         = lockState;
-    LockOpCredentials userCredential[] = { { CredentialTypeEnum::kPin, uint16_t(credentialIndex) } };
-    auto userCredentials               = MakeNullable<List<const LockOpCredentials>>(userCredential);
-    DoorLockServer::Instance().SetLockState(mEndpointId, mLockState, opSource, MakeNullable(static_cast<uint16_t>(userIndex + 1)),
-                                            userCredentials, fabricIdx, nodeId);
+    if (currentAction.moving == true)
+    {
+        return false;
+    }
+
+    currentAction.moving          = true;
+    currentAction.endpointId      = mEndpointId;
+    currentAction.lockState       = lockState;
+    currentAction.opSource        = opSource;
+    currentAction.userIndex       = MakeNullable(static_cast<uint16_t>(userIndex + 1));
+    currentAction.credentialIndex = static_cast<uint16_t>(credentialIndex);
+    currentAction.fabricIdx       = fabricIdx;
+    currentAction.nodeId          = nodeId;
+
+    // simulate 3s lock movement duration
+    chip::DeviceLayer::SystemLayer().StartTimer(chip::System::Clock::Seconds16(3), OnLockActionCompleteCallback, nullptr);
```

Comment:
We may want to consider a value lower than 3 seconds since these delays add up in CI backend testing.

Reply:
I've adapted the tests accordingly.
But I can reduce the delay as well @bzbarsky-apple ?

Reply:
Changed to 0s